orangecontrib/geo/widgets/owgeotransform.py

```diff
@@ -0,0 +1,153 @@
+        data = Input("Data", Table)
+
+    class Outputs:
+        data = Output("Data", Table)
+
+    settingsHandler = DomainContextHandler()
+    attr_lat = settings.ContextSetting(None)
+    attr_lon = settings.ContextSetting(None)
+    from_idx = settings.Setting("Slovenia 1996 / Slovene National Grid")
+    to_idx = settings.Setting("WGS 84")
+
+    want_main_area = False
+    resizing_enabled = False
+
+    EPSG_CODES = get_projections()
+
+    class Error(OWWidget.Error):
+        no_lat_lon_vars = Msg(
+                "Data has no latitude and longitude variables.")
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+        self.lat = None
+        self.lon = None
+        self.transform_from = None
+        self.transform_to = None
+
+        box = gui.vBox(self.controlArea, box="Coordinates:")
+
+        self.variable_model = DomainModel(
+            order=DomainModel.MIXED, valid_types=(ContinuousVariable,))
+
+        self.lat_combo = gui.comboBox(box, self, "attr_lat",
+                                      label="Latitude:",
+                                      contentsLength=100,
+                                      searchable=True,
+                                      model=self.variable_model,
+                                      orientation=Qt.Horizontal)
+
+        self.lon_combo = gui.comboBox(box, self, "attr_lon",
+                                      label="Longitude:",
+                                      contentsLength=100,
+                                      searchable=True,
+                                      model=self.variable_model,
+                                      orientation=Qt.Horizontal)
+
+        box = gui.vBox(self.controlArea, box="Transformation:")
+
+        self.from_combo = gui.comboBox(
+            box, self, "from_idx",
+            contentsLength=100,
+            searchable=True,
+            items=list(self.EPSG_CODES.keys()),
+            label="From:", orientation=Qt.Horizontal,
+            sendSelectedValue=True
+        )
+
+        self.to_combo = gui.comboBox(
+            box, self, "to_idx",
+            contentsLength=100,
+            searchable=True,
+            items=list(self.EPSG_CODES.keys()),
+            label="To:", orientation=Qt.Horizontal,
+            sendSelectedValue=True
+        )
+
+        self.commit_button = gui.button(self.controlArea, self, "&Commit",
+                                        self.apply)
+
+    def init_attr_values(self):
+        self.Error.clear()
+        lat, lon = None, None
+        if self.data is not None:
+            lat, lon = find_lat_lon(self.data, filter_hidden=True)
+            if lat is None or lon is None:
+                # we either find both or we don't have valid data
+                self.Error.no_lat_lon_vars()
+                self.data = None
+                lat, lon = None, None
+
+        self.variable_model.set_domain(self.data.domain if self.data else None)
```

Comment:
This does not happen, if the data is None, so the Lat/Lon comboboxes retain the content when the data signal is disconnected.

Reply:
This was copied from GeoMap and Choropleth. 🤷‍♀️ Is it wrong there as well?

Reply:
It's not wrong, it's just not nice that when there is no data, the combos are full.